### PR TITLE
migrate to v3

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

WIP - still fixing tests

### CODE REVIEW

**Comment by Ethan-Arrowood** [comment]
tests seem to be failing because other plugins in it aren't update to 3 yet... will revisit

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: ['14', '12', '10']
+      name: Node ${{ matrix.node }}
+      steps:
+        - uses: actions/checkout@v1
```
Comment (Eomm) [inline .github/workflows/ci.yml:12]:
```suggestion
        - uses: actions/checkout@v2
```

File: auth.js [hunk 2]
```diff
@@ -117,4 +117,4 @@ function auth (functions, opts) {
   }
 }
 
-module.exports = fp(checkAuth, '>=0.13.1')
+module.exports = fp(checkAuth, '>=3')
```
Comment (mcollina) [inline auth.js:120]:
```suggestion
module.exports = fp(checkAuth, '3.x')
```

**Comment by mcollina** [comment]
@Ethan-Arrowood it seems CI is failing.

**Comment by mcollina** [comment]
Can you add dependabot and drop travis?

**Comment by Ethan-Arrowood** [comment]
@mcollina iirc those have to be configured by an admin. I don't have access to the repo settings

**Comment by mcollina** [comment]
Nope, they are already enabled. Just add the config file.

**Comment by Ethan-Arrowood** [comment]
Unless I'm missing something there doesn't seem to be any need for a dependabot file. (And none of the other fastify repos that I checked have one 🤷‍♀️) Deleting travis now tho

**Comment by mcollina** [comment]
> Unless I'm missing something there doesn't seem to be any need for a dependabot file.

It's enabled automatically if you add one. So, we'll need to add that config file to this repo. I'll do it for this one.

**Comment by mcollina** [comment]
All tests were failing, and CI was not configured correctly and did not run. I'll be fixing this.

**Comment by Ethan-Arrowood** [comment]
Oh I see the dependabot thing now, my mistake. I can revisit the tests and ci if you haven't already done so. Just let me know

**Pull request merged**